LaTeX/LaTeX.sublime-syntax

```diff
@@ -4,554 +4,1545 @@
+        1: support.function.ensuremath.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_scope: meta.function.ensuremath.latex
+        - meta_content_scope: meta.environment.math.inline.ensuremath.latex
+        - match: '\}'
+          scope: punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: math-content
+
+    - match: (\\\()
+      scope: string.other.math.latex
       captures:
-        1: support.function.be.latex
-        2: punctuation.definition.function.latex
-        3: punctuation.definition.arguments.latex
-        4: variable.parameter.function.latex
-        5: punctuation.definition.arguments.latex
+        1: punctuation.definition.string.begin.latex
       push:
-        - meta_scope: meta.function.environment.list.latex
-        - match: '((\\)end)(\{)(\4)(\})(?:\s*\n)?'
-          captures:
-            1: support.function.be.latex
-            2: punctuation.definition.function.latex
-            3: punctuation.definition.arguments.latex
-            4: variable.parameter.function.latex
-            5: punctuation.definition.arguments.latex
+        - meta_scope: meta.environment.math.inline.paren.latex
+        - match: (\\\))
+          scope: string.other.math.latex punctuation.definition.string.end.latex
           pop: true
-        - include: $top_level_main
-    - match: '(?:\s*)((\\)begin)(\{)(\w+[*]?)(\})'
+        - include: math-content
+
+  block-math:
+    - match: \$\$
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: meta.environment.math.block.dollar.latex
+        - match: \$\$
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+        - include: math-content
+
+    - match: '(\\\[)'
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: meta.environment.math.block.bracket.latex
+        - match: '(\\\])'
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+        - include: math-content
+
+    - match: '((\\)begin)(\{)\s*((?:align|alignat|aligned|alignedat|displaymath|eqnarray|equation|flalign|gather|gathered|math|multline|x?xalignat|split)\*?)\s*(\})'
       captures:
         1: support.function.be.latex
-        2: punctuation.definition.function.latex
-        3: punctuation.definition.arguments.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
         4: variable.parameter.function.latex
-        5: punctuation.definition.arguments.latex
+        5: punctuation.definition.group.brace.end.latex
       push:
-        - meta_scope: meta.function.environment.general.latex
-        - match: '((\\)end)(\{)(\4)(\})(?:\s*\n)?'
+        - meta_scope: meta.environment.math.block.be.latex
```

Comment:
I think it would be nice to highlight all math content in a specific color. But nonetheless also math keybindings should only be applied in the body not inside `\begin{...}` and `\end{...}`.